src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/GraphSymbolUtils.cs

```diff
@@ -143,18 +143,161 @@ private static IEnumerable<INamedTypeSymbol> GetAllViewTypesFromPXGraphOrPXGraph
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates.Add(d.Item2.Name, d);
+                }
+            }
+
+            void AddDelegatesFromGraphExtension(Dictionary<string, (MethodDeclarationSyntax, IMethodSymbol)> delegates, ITypeSymbol graphExt)
+            {
+                IEnumerable<(MethodDeclarationSyntax, IMethodSymbol)> dels = graphExt.GetViewDelegatesFromGraphOrGraphExtension(views, pxContext, cancellation);
+
+                foreach ((MethodDeclarationSyntax, IMethodSymbol) d in dels)
+                {
+                    delegates[d.Item2.Name] = d;
+                }
+            }
+        }
+
+        private static IEnumerable<T> GetViewInfoFromGraphExtension<T>(ITypeSymbol graphExtension, PXContext pxContext,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphViewInfo,
+            Action<Dictionary<string, T>, ITypeSymbol> addGraphExtensionViewInfo)
+        {
+            IEnumerable<T> empty = Enumerable.Empty<T>();
+
+            if (!graphExtension.InheritsFrom(pxContext.PXGraphExtensionType) || !graphExtension.BaseType.IsGenericType)
+            {
+                return empty;
+            }
+
+            INamedTypeSymbol baseType = graphExtension.BaseType;
+            Dictionary<string, T> infoByView = new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase);
+
+            for (int i = baseType.TypeArguments.Length - 1; i >= 0; i--)
+            {
+                ITypeSymbol argType = baseType.TypeArguments[i];
+
+                if (i == baseType.TypeArguments.Length - 1)
+                {
+                    if (!argType.IsPXGraph(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphViewInfo(infoByView, argType);
+                }
+                else
+                {
+                    if (!argType.IsPXGraphExtension(pxContext))
+                    {
+                        return empty;
+                    }
+
+                    addGraphExtensionViewInfo(infoByView, argType);
+                }
+            }
+
+            addGraphExtensionViewInfo(infoByView, graphExtension);
+
+            return infoByView.Values;
+        }
+
+        private static IEnumerable<(MethodDeclarationSyntax Node, IMethodSymbol Symbol)> GetViewDelegatesFromGraphOrGraphExtension(
+            this ITypeSymbol graphOrExtension, IEnumerable<ISymbol> views, PXContext pxContext, CancellationToken cancellation)
+        {
+            foreach (ISymbol member in graphOrExtension.GetMembers())
+            {
+                cancellation.ThrowIfCancellationRequested();
+
+                if (!(member is IMethodSymbol method))
+                {
+                    continue;
+                }
+
+                bool correctName = views.Any(v => v.Name.Equals(member.Name, StringComparison.OrdinalIgnoreCase));
```

Comment:
@metallmonster what is the average views count in graph? Is it sensible to create hashset from the view names for this search?

Reply:
@SENya1990 I think there are a few dozens of views in a graph in general so it is not necessary to use a hashset here.

Reply:
@metallmonster a few dozens means around 30-40? I thought it was much less. Actuall Hashset is more perfomant in this case but the difference is not so big so decide yourself which is better.
According to this benchmark hashset becomes more performant  at about 20 items:
https://stackoverflow.com/questions/150750/hashset-vs-list-performance